### ISSUE TITLE
Staff of the Clouds giving infinite cloud blocks

### ISSUE THREAD

**Issue by KikiCur** [ShetiPhian/Localizations-and-Issues#160]
hi, :) lovely mod, happily working my way through it and multistorage

Playing Forever Stranded Lost Souls version 1.0.3.10-B3
so Forge v. 14.23.2.2618
Terraqueous v. 1.12.0-1.4.9

The Staff of the Clouds doesn't use the water blocks it contains when it makes light cloud blocks. Thanks for your time. :)

**Comment by ShetiPhian** [comment]
I'm not having the issue here, though I haven't tested in that pack so it may be a conflict.
Incoming question barrage :)

Are you using the staff to place the blocks in world, or pulling from a crafting grid?
InWorld > are you in creative mode? (its infinite while in creative)
CraftingGrid > is it a mod added workbench? does it work correct from you inventory?

Is this server or single player?
If Server > does it work in single player?

**Comment by KikiCur** [comment]
hi, sorry, thanks for reply
nope, I'm using the staff both in my personal 4 block crafting grid, and I also used it in the tinker's table beside a storage crate >_< :D which it filled up with light cloud blocks
my staff has 16 buckets of water in it, but a friend also tested it and only has 8 buckets in his same result, infinite clouds, single player for both of us

**Comment by ShetiPhian** [comment]
Just checked in a MultiMC instance (was in my build environment previously)
Personal Grid, Vanilla Crafting Table, and my Craft Bench all worked  correctly. (didn't have Tinker's to test)

Its odd that the personal grid works but not the rest.
Fluid is changed in Item.getContainerItem something in that pack is either stopping the call or ignoring the result.

I'll need to look at the pack.
Is there any recipes that need a water bucket? it also uses that method to turn into an empty bucket.

**Comment by ShetiPhian** [comment]
Just downloaded the pack, saw the issue first hand.
I am so dumb -_-

I think I know exactly where the bug is, and It should be reproducible in my development environment.

**Comment by ShetiPhian** [comment]
This was 100% me.
I failed to notice you only had enough water for the craft, leaving the staff empty.

This bug is because I didn't fully test another bug fix I did.
Summery: 
Received a crash log with a NullPointerError;  Looked at code, sure enough I was using a variable without checking it for null first;  Added a null check and called it a day.
As it turns out that new null check made the staff unable to become empty.

I have now properly fixed the old bug, that in turn corrects this issue.
